src/ranged.cpp

```diff
@@ -161,19 +161,79 @@ bool player::handle_gun_damage( item &it, int shots_fired )
 {
     if( !it.is_gun() ) {
         debugmsg( "Tried to handle_gun_damage of a non-gun %s", it.tname() );
+        shots_fired = shots_fired + 0; // left this here in case someone needs shots_fired in the future.
         return false;
     }
-
+    int bpjamoccurredthisturn = 0;
     const auto &curammo_effects = it.ammo_effects();
     const cata::optional<islot_gun> &firing = it.type->gun;
+    const std::string thistimefired = to_string_time_of_day( calendar::turn );
+    // Here we check for a chance for the weapon to suffer a misfire due to
+    // being dirty (blackpowder fouling effect only; typical dirt accumulation
+    // from smokeless powder is not currently modeled by this effect).
+    // Can damage gun, but not enough to push the weapon beyond 'shattered'.
+    // Check, but not accumulation, will happen even when firing normal rounds,
+    // so you can screw up your MP5 by firing blackpowder rounds and the
+    // effects will be in force even when firing retail ammo, though the gun
+    // won't get MORE foul.
+
+    // causes failure to cycle if weapon is fired too quickly:
+    if( it.last_fired == thistimefired && !it.has_flag( "PUMP_ACTION" ) &&
```

Comment:
Why a string? Why are you not just using `time_point`. Or even just `int` representing the turn number (get it via `to_turn<int>( calendar::turn )`. But converting the current time to a (human readable string!) makes no sense.

Btw. `last_fired` really should be an item variable.

Reply:
I don't get the comment re: last_fired.